libs/libcommon/src/libcommon/viewer_utils/asset.py

```diff
@@ -35,25 +38,32 @@ def create_image_file(
     filename: str,
     image: Image.Image,
     format: str,
-    storage_client: StorageClient,
+    storage_client: "StorageClient",
 ) -> ImageSource:
+    # We use a placeholder revision that will be replaced later by the
+    # dataset_git_revision of cache responses when the data will be accessed.
+    # This is useful to allow moving files to a newer revision without having
+    # to modify the cached rows content.
     object_key = storage_client.generate_object_key(
         dataset=dataset,
-        revision=revision,
+        revision=DATASET_GIT_REVISION_PLACEHOLDER,
         config=config,
         split=split,
         row_idx=row_idx,
         column=column,
         filename=filename,
     )
-    if storage_client.overwrite or not storage_client.exists(object_key):
+    path = replace_dataset_git_revision_placeholder(object_key, revision=revision)
+    if storage_client.overwrite or not storage_client.exists(path):
         image = ImageOps.exif_transpose(image)  # type: ignore[assignment]
         buffer = BytesIO()
         image.save(fp=buffer, format=format)
         buffer.seek(0)
-        with storage_client._fs.open(storage_client.get_full_path(object_key), "wb") as f:
+        with storage_client._fs.open(storage_client.get_full_path(path), "wb") as f:
             f.write(buffer.read())
-    return ImageSource(src=storage_client.get_url(object_key), height=image.height, width=image.width)
+    return ImageSource(
+        src=storage_client.get_url(object_key, revision=revision), height=image.height, width=image.width
+    )
```

Comment:
A few lines later, `get_url` applies the right revision if there is an URL preparator associated to the storage client.

In practive:
- /rows, /search and /filter have an URL preparator that inserts the revision and signs the urls from the parquet/duckdb data
- split-first-rows worker has no URL preparator: URLs are stored in the cached as unprepared (i.e. with the revision placeholder and unsigned)
- /first-rows has an URL preparator that inserts the revision and signs the urls that come form the cache